test/integration/targets/module_defaults/collections/ansible_collections/testns/testcoll/meta/runtime.yml

```diff
@@ -1,3 +1,38 @@
+plugin_routing:
+  action:
+    # Test for action plugin redirects for module_defaults.
+    # Each module_defaults entry is first resolved as an action plugin, and if it does not exist, it is resolved as a module.
+    # If multiple modules use the same action plugin redirect, module defaults for any of the modules as redirected actions will apply.
+    # Since they will all resolve to the same thing, last defined wins.
+    module_uses_action_defaults:
+      redirect: testns.testcoll.eos
+  modules:
+    # Any module_defaults for testns.testcoll.module will not apply to a module_uses_action_defaults task:
+    #
+    # module_defaults:
+    #   testns.testcoll.module:
+    #     option: value
+    #
+    # But defaults for testns.testcoll.module_uses_action_defaults or testns.testcoll.eos will:
+    #
+    # module_defaults:
+    #   testns.testcoll.module_uses_action_defaults:
+    #     option: value
+    #   testns.testcoll.eos:
+    #     option: defined_last_i_win
+    module_uses_action_defaults:
+      redirect: testns.testcoll.module
+
+    # Not "eos_facts" to ensure TE is not finding handler via prefix
+    # eosfacts tasks should not get eos module_defaults (or defaults for other modules that use eos action plugin)
+    eosfacts:
```

Comment:
I would add another test with eos_facts/ios_facts to see the 'other matching path' working

Reply:
can also be added to test_defaults with collections: , so we get task with the non fqcn name

Reply:
I added better tests for ios_facts with the collections keyword. Now the test suite verifies that calling the task with either iosfacts or ios_facts will get module_defaults entries `testns.testcoll.iosfacts`/`testns.testcoll.ios_facts`.

I also removed the changes to make more stuff rely on the prefix magic. Now this is fixed only via toggle, so the runtime.yml changed a bit to match the tests' expectations.